### DISCUSSION TITLE
'where' matrix iterative updates for applicability/skip-logic

### DISCUSSION THREAD

**Discussion by Mayari** [amices/mice#342] [Impute, analyse and pool]
Hi, I have been making use of interactions and of the 'where' matrix to ensure consistent imputations in the presence of items whose applicability depends on other items (for example: did you move?=1/0, if yes, at what age?), which works smoothly except for cases where the applicability indicator itself (did you move?) is also missing for some entries. In those cases, because both types of items are missing, the entries in the "where" matrix are left as "TRUE" and implausible imputations for the applicability-dependent variables are created. 
So, I was wondering **if it would be possible to request that some columns of the 'where' matrix be updated on each iteration based on the values of other user-specified variables (like the applicability indicators)**. So that, for example, if the user specifies that the "age at time of move" TRUE/FALSE column in the "where" matrix depends on the imputed "did you move?" variable, then for entries that get imputed as 0 (didn't move) the entries in the 'where' matrix for "age at the time of move" would be overwritten with FALSE and no implausible values would be imputed.
This would be very useful in contexts where we have many databases to impute with large numbers of variables and many types of applicability dependencies, as opposed to manually overwriting implausible values and creating diagnostics plots after overwriting implausible imputations.

Thank you for your consideration and for all the incredible work put into the mice package, which already does wonders and allows for so much flexibility based on user/data specific needs.

**Comment by stefvanbuuren** [comment]
It seems to me that you can solve your case by conditional imputation. See #258 for two methods. Hope this helps.
